### ISSUE TITLE
[FEATURE]: Highlight the staff that have the "shortest finish time" in the staff selection page

### ISSUE THREAD

**Issue by chunyinng078** [COMP3122-2324-Project/Simpsyber#10]
In the select staff page for assigning tasks for staff, it would be beneficial to have a "Highlight" feature to directly indicate/point out which staff have the best efficiency (or shortest finish time) of the task. It would be handy for making decisions, as players/users don't have to click on each staff member to see who could complete the task in the shortest time.

Like there were pair of "Software Engineer" or "Tester," we could intuitively choose the "best" staff (the best "Software Engineer" or "Tester," ) for a specific task. Another example is that when there is a "QA" task to do but we are running out of "tester," then we could find a best "replacement" for "tester" easily.

Although we all know that watching the "details" of the staff is essential, what we see is the finish time after all.
Therefore, this enhancement would be helpful for users and even improve the user experience.

Before:
![image](https://github.com/COMP3122-2324-Project/Simpsyber/assets/95467201/6799a2ae-d4a4-4925-a2b6-3eec52c4c93e)

After:
![image](https://github.com/COMP3122-2324-Project/Simpsyber/assets/95467201/e67a161f-b2a9-4f90-8a45-cdd35ac9dd52)

**Comment by ckw-polyu** [comment]
I agree with you, the same position staff will have different finish time. Each time, I need to click it to check who is faster, which is inconvenient.

**Comment by less2771** [comment]
I also agree that a recommended staff should be highlighted on staff selection function. For some of the staffs who are occupied by the task, I suggested that moved those staffs to the back of the list that reduced the time of searching the suitable staff for the new tasks, or separated those working or available staff on the staff selection function to make the function clear and user-friendly.

**Comment by chunyinng078** [comment]
@less2771  "Move those staff to the back of the list" may be a good suggestion! If the staff that cannot be assigned were at the back, users could be more focused on the staff that could be assigned, which could provide a better user experience.

Based on my issue regrading the "ease of assigning tasks for staff" and to elevate your suggestion, I suggest we remove the "staff that could not be assigned" from the list since they would not be considered, which is redundant. Then users could be even more focused on the staff that could be assigned. Therefore, the function of assigning tasks would more clear and easy to use, providing an even better user experience!

**Comment by vincentforive** [comment]
Highlighting staff with less time is a good suggestion, and on that basis I would suggest that game developers could automatically sort staff based on development time and exclude unavaliable staff from the sequence.

Or even provide a button for the user to automatically select the fastest employee, e.g. let the user directly select the fastest employee

<img width="753" alt="image" src="https://github.com/COMP3122-2324-Project/Simpsyber/assets/84238772/ad36366c-f1db-4eba-9b5a-c71361acc0b9">

This saves a lot of clicks.

Originally
Select Job -> Select Employee -> Click to Confirm

has changed to
Select Job->Select Fastest Employee

This speeds up the pace of the game considerably

**Comment by gcctbilly** [comment]
I think in addition to listing those who require the shortest time, it would be best to list the working hours of all employees at once in the selection. In this way, the time gap required by employees on this task can be seen more intuitively. Because sometimes, in order to minimize the waiting time, we may give priority to the tasks that an employee is best at around all the tasks (the time difference between him and others is the largest among all other tasks).

**Comment by bronyazzy** [comment]
I agree with you, but I think it would also be possible to sort the employees hours by worked  . You can even use the "Stable Marriage Problem" algorithm to select when there are multiple free employees and multiple free tasks.
